WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -1621,7 +1621,11 @@ private extension EditableOrderViewModel {
         }
 
         // Return a new input with the new quantity but with the same item id to properly reference the update.
-        return OrderSyncProductInput(id: item.itemID, product: product, quantity: quantity, discount: discount ?? currentDiscount(on: item))
+        return OrderSyncProductInput(id: item.itemID,
+                                     product: product,
+                                     quantity: quantity,
+                                     discount: discount ?? currentDiscount(on: item),
```

Comment:
I think there's an issue when we update the `OrderSyncProductInput` here if there's a discount, since this is calculated as product total - product subtotal of the item. With no discount we can see the rounding price causing a difference of $0.04, but if I add a discount, this gap increases to $0.17 (the discount being only $1.00).

| 10 items (no discount) | 10 items (1 discounted) |
|--------|--------|
| ![Simulator Screenshot - iPhone 15 - 2023-10-26 at 10 58 02](https://github.com/woocommerce/woocommerce-ios/assets/3812076/543f2356-cf35-42ed-905b-fc1e1233f3de) | ![Simulator Screenshot - iPhone 15 - 2023-10-26 at 10 58 23](https://github.com/woocommerce/woocommerce-ios/assets/3812076/61e1b219-e2dd-49db-a17f-95c55e3a536f) |